cmd/rule-evaluator/main.go

```diff
@@ -951,10 +1051,26 @@ func (e *ruleEvaluator) Query(ctx context.Context, q string, t time.Time) (promq
 }
 
 func (e *ruleEvaluator) Run() {
-	e.rulesManager.Run()
+	for {
+		// Copy the rule-manager before running, so we don't hold the lock.
+		e.mtx.Lock()
+		oldRuleManager := e.rulesManager
```

Comment:
I will think about perhaps if there's anything to simplify here

Reply:
Note: This was split to other PR: https://github.com/GoogleCloudPlatform/prometheus-engine/pull/1129